plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/AppEngineStandardWizardPage.java

```diff
@@ -64,23 +66,37 @@ public void createControl(Composite parent) {
     PlatformUI.getWorkbench().getHelpSystem().setHelp(container,
         "com.google.cloud.tools.eclipse.appengine.newproject.NewProjectContext"); //$NON-NLS-1$
 
-    ModifyListener pageValidator = new PageValidator();
 
-    // Java package name
-    Label packageNameLabel = new Label(container, SWT.NONE);
-    packageNameLabel.setText(Messages.getString("java.package")); //$NON-NLS-1$
-    javaPackageField = new Text(container, SWT.BORDER);
-    GridData javaPackagePosition = new GridData(GridData.FILL_HORIZONTAL);
-    javaPackagePosition.horizontalSpan = 2;
-    javaPackageField.setLayoutData(javaPackagePosition);
-    javaPackageField.addModifyListener(pageValidator);
+    createPackageField(container);
 
     // Manage APIs
     appEngineLibrariesSelectorGroup = new AppEngineLibrariesSelectorGroup(container);
 
     Dialog.applyDialogFont(container);
   }
 
+  // Java package name
+  private void createPackageField(Composite container) {
+    
+    Composite composite = new Composite(container, SWT.NONE);
+    GridData compositePosition = new GridData(GridData.FILL_HORIZONTAL);
+    compositePosition.horizontalSpan = 2;
+    composite.setLayoutData(compositePosition);
```

Comment:
By setting the layout-data to a `GridData`, you're assuming that `container` has a `GridLayout`.  That's not a good practice, unless it's documented explicitly. For example, if we decide to add some other config items (like placement of the `WEB-INF`) and group them together in a `Group` that uses a `RowLayout`, then this code breaks.

IMHO, a better approach would be to create the composite in `createControl()`, and pass it in here with the documentation saying that it's laid-out with a GridLayout with 2 columns.